Wikipedia/UI-V5/WMFNearbyViewModel.m

```diff
@@ -0,0 +1,193 @@
+}
+
+- (void)setSite:(MWKSite* __nonnull)site {
+    if (WMF_EQUAL(self.site, isEqualToSite:, site)) {
+        return;
+    }
+    _site = site;
+    [self fetchTitlesForLocation:self.locationSearchResults.location];
+}
+
+- (void)setLocationSearchResults:(WMFLocationSearchResults* __nullable)locationSearchResults {
+    if (WMF_EQUAL(self.locationSearchResults, isEqual:, locationSearchResults)) {
+        return;
+    }
+    [self stopUpdatingAllResults];
+    _locationSearchResults = locationSearchResults;
+}
+
+#pragma mark - Result Updates
+
+- (void)autoUpdateResultAtIndex:(NSUInteger)index {
+    [self.autoUpdatingIndexes addIndex:index];
+    DDLogInfo(@"Tracking headings for %lu indexes", self.autoUpdatingIndexes.count);
+    [self updateBearingAtIndex:index];
+    [self updateDistanceAtIndex:index];
+}
+
+- (void)stopUpdatingResultAtIndex:(NSUInteger)index {
+    [self.autoUpdatingIndexes removeIndex:index];
+    DDLogInfo(@"Tracking headings for %lu indexes", self.autoUpdatingIndexes.count);
+    // TODO: toggle heading tracking based on whether or not we have indexes to track
+}
+
+- (void)stopUpdatingAllResults {
+    [self.autoUpdatingIndexes enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL* stop) {
+        [self stopUpdatingResultAtIndex:idx];
+    }];
+}
+
+- (void)updateBearingAtIndex:(NSUInteger)index {
+    MWKLocationSearchResult* result = self.locationSearchResults.results[index];
+    result.bearingToLocation =
+        [self.locationSearchResults.location wmf_bearingToLocation:result.location
+                                                 forCurrentHeading:self.locationManager.lastHeading];
+}
+
+- (void)updateDistanceAtIndex:(NSUInteger)index {
+    MWKLocationSearchResult* result = self.locationSearchResults.results[index];
+    result.distanceFromUser = [result.location distanceFromLocation:self.locationManager.lastLocation];
+}
+
+#pragma mark - Fetch
+
+- (void)startUpdates {
+    [self.locationManager restartLocationMonitoring];
+    // TODO: check if authorized. if not, fail w/ an error
+}
+
+- (void)stopUpdates {
+    [self.locationManager stopMonitoringLocation];
+    self.locationSearchResults = nil;
+    // TODO: cancel last search request
+}
+
+- (void)fetchTitlesForLocation:(CLLocation* __nullable)location {
+    if (!location) {
+        return;
+    } else if ([self.locationSearchResults.location distanceFromLocation:location] < 500
+               && [self.locationSearchResults.searchSite isEqualToSite:self.site]) {
+        /*
+           How often we search titles for a location is separate from how often the location updates.
+           See WMFLocationManager for distanceFilter settings.
+         */
+        DDLogVerbose(@"Not fetching nearby titles for %@ since it is too close to previously fetched location: %@.",
+                     location, self.locationSearchResults.location);
+        return;
+    }
+
+    NSAssert(![self.locationSearchResults.location wmf_isVeryCloseToLocation:location]
```

Comment:
We should be deliberate in our assertions (really all code). 

If the condition isn't possible, then don't include it. If it is possible, then we should also handle this gracefully at run time (Another return method).